pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -433,6 +433,35 @@ type GarbageCollector struct {
 	propagator  *Propagator
 }
 
+// TODO: make special List and Watch function that removes fields other than
+// ObjectMeta.
+func gcListWatcher(client *dynamic.Client, resource unversioned.GroupVersionResource) *cache.ListWatch {
+	return &cache.ListWatch{
+		ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+			// APIResource.Kind is not used by the dynamic client, so
+			// leave it empty. We want to list this resource in all
+			// namespaces if it's namespace scoped, so leave
+			// APIResource.Namespaced as false is all right.
+			apiResource := unversioned.APIResource{Name: resource.Resource}
+			// The default parameter codec used by the dynamic client cannot
+			// encode api.ListOptions.
+			// TODO: api.ParameterCodec doesn't support thirdparty objects.
+			// We need a generic parameter codec.
+			return client.Resource(&apiResource, api.NamespaceAll).List(&options, api.ParameterCodec)
```

Comment:
Technically list options should be in the normal schemes, but even in
request.go it made more sense to let the parameter codec vary while the
fundamental codec remained the same.

On Thu, Jun 2, 2016 at 7:35 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/controller/garbagecollector/garbagecollector.go
> <https://github.com/kubernetes/kubernetes/pull/26493#discussion_r65637027>
> :
>
> > @@ -433,6 +433,35 @@ type GarbageCollector struct {
> >  	propagator  *Propagator
> >  }
> >
> > +// TODO: make special List and Watch function that removes fields other than
> > +// ObjectMeta.
> > +func gcListWatcher(client *dynamic.Client, resource unversioned.GroupVersionResource) *cache.ListWatch {
> > +	return &cache.ListWatch{
> > +		ListFunc: func(options api.ListOptions) (runtime.Object, error) {
> > +			// APIResource.Kind is not used by the dynamic client, so
> > +			// leave it empty. We want to list this resource in all
> > +			// namespaces if it's namespace scoped, so leave
> > +			// APIResource.Namespaced as false is all right.
> > +			apiResource := unversioned.APIResource{Name: resource.Resource}
> > +			// The default parameter codec used by the dynamic client cannot
> > +			// encode api.ListOptions.
> > +			// TODO: api.ParameterCodec doesn't support thirdparty objects.
> > +			// We need a generic parameter codec.
> > +			return client.Resource(&apiResource, api.NamespaceAll).List(&options, api.ParameterCodec)
>
> Just saw krousey's reply. He's right.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26493/files/fd1c0abc1da522676ac1800b5f482a4db3d172cc#r65637027>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_pz2WXtklAs94aLKr3Y3C8Dm3hvULks5qH2jUgaJpZM4IpLXf>
> .
>